### PR TITLE
changelog: add breaking change section and reorg

### LINKED ISSUE #597: CHANGELOG: breaking changes section
i think we should separate out the breaking changes and highlight them for our users

_Originally posted by @williballenthin in https://github.com/fireeye/capa/pull/593#r641704281_

### PR DESCRIPTION
**Pull request by mr-tz**
closes #597 

### Checklist

- [ ] No CHANGELOG update needed

- [x] No new tests needed

- [x] No documentation update needed

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -107,29 +110,27 @@ It includes many new rules, including all new techniques introduced in MITRE ATT
 - build: use Python 3.8 for PyInstaller to support consistently running across multiple operating systems including Windows 7 #505 @mr-tz
 - main: correctly match BB-scope matches at file scope #605 @williballenthin
 
-### Changes
+### Breaking Changes
 
 - py3: drop Python 2 support #480 @Ana06
-- deps: bump ruamel yaml parser to 0.17.4 #519 @williballenthin
```
Comment (mr-tz) [inline CHANGELOG.md:113]: removed as I think we should not track these here (related #550)

File: CHANGELOG.md [hunk 2]
```diff
@@ -107,29 +110,27 @@ It includes many new rules, including all new techniques introduced in MITRE ATT
 - build: use Python 3.8 for PyInstaller to support consistently running across multiple operating systems including Windows 7 #505 @mr-tz
 - main: correctly match BB-scope matches at file scope #605 @williballenthin
 
-### Changes
+### Breaking Changes
```
Comment (williballenthin) [inline CHANGELOG.md:113]: i think we should have a "breaking changes" section that is near the top, near "new features", and then keep the (non-breaking) "changes" section here. thoughts?

Comment (mr-tz) [reply CHANGELOG.md:113]: I've moved these entries to the other sections. Will move `Breaking Changes` to the top.

File: CHANGELOG.md (hunk 1 above)
Comment (mr-tz) [reply CHANGELOG.md:113]: none of the other entries is lost, they're just moved to different sections

sorry, for not making that clearer inititally

File: CHANGELOG.md (hunk 2 above)
Comment (mr-tz) [reply CHANGELOG.md]: IMO all changes can be grouped into other sections as "everything changes" - 🧙

File: CHANGELOG.md [hunk 3]
```diff
@@ -101,27 +114,14 @@ It includes many new rules, including all new techniques introduced in MITRE ATT
...
 - build: use Python 3.8 for PyInstaller to support consistently running across multiple operating systems including Windows 7 #505 @mr-tz
 - main: correctly match BB-scope matches at file scope #605 @williballenthin
 
-### Changes
-
-- py3: drop Python 2 support #480 @Ana06
-- deps: bump ruamel yaml parser to 0.17.4 #519 @williballenthin
+### Documentation
 - explorer: explain how to install IDA 7.6 patch to enable the plugin #528 @williballenthin
 - explorer: document IDA 7.6sp1 as alternative to the patch #536 @Ana06
```
Comment (mr-tz) [inline CHANGELOG.md:124]: maybe the explorer could have its own additional section

File: CHANGELOG.md [hunk 4]
```diff
@@ -7,13 +7,26 @@ It includes many new rules, including all new techniques introduced in MITRE ATT
...
+- linter: summarize results at the end #571 @williballenthin
+- linter: check for `or` with always true child statement, e.g. `optional`, colors #348 @mr-tz
+
+### Breaking Changes
+
+- py3: drop Python 2 support #480 @Ana06
+- meta: added `library_functions` field, `feature_counts.functions` does not include library functions any more #562 @mr-tz
+- json: results document now contains parsed ATT&CK and MBC fields instead of canonical representation #526 @mr-tz
+- json: record all matching strings for regex #159 @williballenthin
+- main: implement file limitations via rules not code #390 @williballenthin
```
Comment (williballenthin) [inline CHANGELOG.md:29]: this looks great!

**Review by williballenthin — APPROVED** [review]

**Comment by mr-tz** [comment]
@Ana06 any comments on this before I resolve conflicts and merge this?

**Review by Ana06 — APPROVED** [review]
LGTM

**Comment by Ana06** [comment]
it closes https://github.com/fireeye/capa/issues/550 as well

**Comment by Ana06** [comment]
I have noticed that in some cases we use the issue number and in some other the PR number. I personally include the PR because it is the code change. Do we want to standardize it (either always issue PR or issue) or is it ok like it is?

**Comment by mr-tz** [comment]
> Do we want to standardize it (either always issue PR or issue) or is it ok like it is?

I'll try to use the PR as well, but I think either is fine. The issue quickly leads to the PR and vice versa.

**Pull request merged**